src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -205,13 +213,38 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 		}
 	}
 
-	public onInsertButtonClick(event: MouseEvent, type: MarkdownButtonType): void {
-		let go = new MarkdownTextTransformer(this._notebookService, this.cellModel, this._instantiationService);
-		let trigger = event.target as HTMLElement;
-		go.transformText(type, trigger);
+	public async onInsertButtonClick(event: MouseEvent, type: MarkdownButtonType): Promise<void> {
+		event.preventDefault();
+		event.stopPropagation();
+		let triggerElement = event.target as HTMLElement;
+		let needsTransform = true;
+		let calloutResult: ILinkCalloutDialogOptions;
+		if (type === MarkdownButtonType.LINK_PREVIEW) {
+			calloutResult = await this.createCallout(type, triggerElement);
+			// If no URL is present, no-op
+			if (!calloutResult.insertLinkUrl) {
+				return;
+			}
+			// If cell edit mode isn't WYSIWYG, use result from callout. No need for further transformation.
+			if (this.cellModel.currentMode !== CellEditModes.WYSIWYG) {
+				needsTransform = false;
+			} else {
+				// Otherwise, re-focus on the output element, and insert the link directly.
+				this.output?.nativeElement?.focus();
+				// Callout is responsible for returning escaped strings
```

Comment:
Hm, if you're going to make this responsibility on the callout then I'd definitely explicitly call it `escapedLinkUrl` and `escapedLinkLabel`.

But in general I think you should leave it up to the caller to do that - because who knows what kind of escaping the caller actually wants? Sure in this case it's probably ok since this is only going to be used in one place mostly likely but it's still a better practice I think